src/main/java/org/apache/mesos/scheduler/recovery/DefaultRecoveryScheduler.java

```diff
@@ -127,7 +127,7 @@ public DefaultRecoveryScheduler(
         return filteredTerminatedTasks;
     }
 
-    private void updateRecoveryPools(Block block) {
+    private void updateRecoveryPools(Optional<Block> block) {
```

Comment:
Subjective: How about switching this parameter to `List<TaskInfo> terminatedTasks` and then having the caller do something like `updateRecoveryPolls(getTerminatedTasks(block))`?